packages/vendure-plugin-sort-by-popularity/src/sort.service.ts

```diff
@@ -1,12 +1,180 @@
-import { Injectable } from '@nestjs/common';
-
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import {
+  ChannelService,
+  Collection,
+  CollectionService,
+  ID,
+  JobQueue,
+  JobQueueService,
+  OrderItem,
+  Product,
+  RequestContext,
+  SerializedRequestContext,
+  TransactionalConnection,
+} from '@vendure/core';
+import { Success } from '../../test/src/generated/admin-graphql';
 @Injectable()
-export class SortService {
-  async setProductPopularity() {
-    // Get orders from pas X months
-    // Combine variant quantities per product
-    // Normalize by total quantity and store as popularity score on product
-    // Get all products and products of child collections for each collection
-    // Use sum of products per collection as popularity score
+export class SortService implements OnModuleInit {
+  private jobQueue: JobQueue<{
+    channelToken: string;
+    ctx: SerializedRequestContext;
+  }>;
+  constructor(
+    private connection: TransactionalConnection,
+    private jobQueueService: JobQueueService,
+    private channelService: ChannelService,
+    private collectionService: CollectionService
+  ) {}
+  async onModuleInit() {
+    this.jobQueue = await this.jobQueueService.createQueue({
+      name: 'calculate-scores',
```

Comment:
Could you name the queue `calculate-popularity-scores`? In Google Cloud it creates a Task Queue in the interface, and then `calculate-popularity-scores` gives the admin a bit more context about what queue this is for